TTKDTrainingApp/src/components/ReactCamera.js

```diff
@@ -0,0 +1,247 @@
+
+  // Ref to store previous recording state in order to prevent calling
+  //  startRecording() more than once
+  const prevRecording = useRef(false);
+  const cameraRef = useRef();
+
+  // Handles starting recording
+  useEffect(() => {
+    if (!prevRecording.current && recording === true) {
+      startRecording();
+    }
+    prevRecording.current = recording;
+  }, [recording, startRecording]);
+
+  // Handles stopping video if back button is pressed
+  useEffect(() => {
+    if (recording) {
+      shouldCancelVideo &&
+        cameraRef.current &&
+        cameraRef.current.stopRecording();
+    } else {
+      shouldCancelVideo && navigation.goBack();
+    }
+  }, [shouldCancelVideo, navigation, recording]);
+
+  // Handles dispatching video and navigation once recording is stopped
+  useEffect(() => {
+    if (recordedVideo) {
+      if (!shouldCancelVideo) {
+        dispatch(addRecordedVideo({...recordedVideo, contentId}));
+        navigation.navigate('VideoWithControls', {
+          recordedVideo: recordedVideo,
+          contentId: contentId,
+        });
+      } else {
+        navigation.goBack();
+      }
+    }
+  }, [shouldCancelVideo, recordedVideo, contentId, dispatch, navigation]);
+
+  const startRecording = useCallback(async () => {
+    // default to mp4 for android as codec is not set
+    try {
+      if (cameraRef.current) {
+        const video = await cameraRef.current.recordAsync({
+          mute: true,
+          maxDuration: maxLength,
+        });
+
+        setRecording(false);
+        setRecordedVideo(video);
+      }
+    } catch (e) {
+      console.error('Recording Failed', e);
+    }
+  }, [maxLength]);
+
+  const handleLoad = response => {
+    const maxDim = 300;
+    const {height, width, orientation} = response.naturalSize;
+    const isPortrait = orientation === 'portrait';
+    const calcHeight = isPortrait
+      ? maxDim
+      : Math.round((height * maxDim) / width);
+    const calcWidth = isPortrait
+      ? Math.round((width * maxDim) / height)
+      : maxDim;
+    setVideoHeight(calcHeight);
+    setVideoWidth(calcWidth);
+    setVideoLoaded(true);
+    console.info('Finished loading');
+  };
+
+  const recordButton = (
+    <TouchableOpacity
+      onPress={() => setShouldShowCountdown(true)}
+      disabled={!videoLoaded}
+      style={{
+        ...styles.capture,
+        ...(videoLoaded ? null : styles.disabledButton),
+      }}>
+      <Text style={styles.text}> RECORD </Text>
+    </TouchableOpacity>
+  );
+
+  return (
+    isFocused && (
+      <View style={styles.container}>
+        <RNCamera
+          ref={cameraRef}
+          style={styles.preview}
+          type={
+            frontCamera
+              ? RNCamera.Constants.Type.front
+              : RNCamera.Constants.Type.back
+          }
+          flashMode={RNCamera.Constants.FlashMode.off}
+          captureAudio={false}
+          androidCameraPermissionOptions={{
```

Comment:
ios asked me to do this automatically btw.... but you probably knew that anyways ahah!

Reply:
`react-native-camera` had some really good boiler plate code that had all the permissions stuff we needed luckily